docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -41,27 +41,34 @@ versions from the sets. Leaving them there is harmless.
 You can use this technique with longer-lived Workflows as well. Just be aware that you might need to
 keep multiple versions of Workers running while open Workflows complete.
 
-You can also use the feature to prevent a buggy codepath from being taken on currently open
-Workflows. You can do this by adding a new version to an existing set and specifying it as
-_compatible_ with an existing version on which no future Workflow Tasks should execute. Because the
-new version processes existing [Event Histories](/workflows/#event-history), it is subject to the
-normal [deterministic constraints](/workflows/#deterministic-constraints), and you might need to use
-one of the [versioning APIs](/workflows/#workflow-versioning).
+You can also use the feature to make compatible changes to, or prevent a buggy codepath from being
+taken, on currently open Workflows. You can do this by adding a new version to an existing set and
+specifying it as _compatible_ with an existing version on which no future Workflow Tasks should
+execute. Because the new version processes existing [Event Histories](/workflows/#event-history), it
+is subject to the normal [deterministic constraints](/workflows/#deterministic-constraints), and you
+might need to use one of the [versioning APIs](/workflows/#workflow-versioning).
 
 The feature also permits you to make incompatible changes to Activity Definitions in conjunction
 with incompatible changes to Workflow Definitions that use those Activities. This works because any
-Activity scheduled by a Workfow on the same Task Queue will, by default, only be dispatched to
+Activity scheduled by a Workflow on the same Task Queue will, by default, only be dispatched to
 Workers that have a compatible build ID with the Workflow that scheduled it. Thus, if you wish
 to change the type signature of an Activity Definition while creating a new incompatible build ID
 for a Worker, you can do so without concern that the Activity will fail to execute on some other
-Worker with an incompatible definition. The same principle applies to Child Workflows.
+Worker with an incompatible definition. The same principle applies to Child Workflows. It's 
+important to keep in mind that "publicly facing" workflows on a versioned Task Queue should not
+change their signature, as this defeats the purpose of workflow-launching clients being oblivious
+to changes in the Workflow Definition. If you need to change the signature of a Workflow, you should
+use an entirely new Task Queue.
```

Comment:
```suggestion
use a different Workflow Type or an entirely new Task Queue.
```
changing the type feels like the normal way to go in that situation to me? but you can switch the order if you like changing the task queue